### PR TITLE
fix: added validation for managed_domain_mappings

### PR DESCRIPTION
**Pull request by Ak-sky**
### Description

Added validation for the variable `managed_domain_mappings`.

### Release required?

- [ ] No release
- [x] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Added `managed_domain_mappings` variable validation.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: modules/app/variables.tf [hunk 1]
```diff
@@ -57,6 +57,10 @@ variable "managed_domain_mappings" {
   description = "Define which of the system managed domain mappings will be setup for the application. Valid values are 'local_public', 'local_private' and 'local'."
   type        = string
   default     = null
+  validation {
```
Comment (akocbek) [inline modules/app/variables.tf:60]: Please add validation in DA as well
https://github.com/terraform-ibm-modules/terraform-ibm-code-engine/blob/da79ed8b9a96d2e9c41fd43df06e4fb2a054bc98/solutions/apps/variables.tf#L97

**Comment by Ak-sky** [comment]
PR is failing for almost every region

```
2024/12/09 11:04:51 Failed GET DETAILS for region eu-gb : The system is currently unable to fulfill the request. Please retry. If the failure persists, contact support and provide the trace value from this error. Full Response: {
```

File: modules/app/variables.tf (hunk 1 above)
Comment (Ak-sky) [reply modules/app/variables.tf:60]: confirmed with @ocofaigh, we need this beacuse projects might use this validation as well.

Comment (Ak-sky) [reply modules/app/variables.tf:60]: Made the changes.

**Review by akocbek — APPROVED** [review]

**Pull request merged**